--- sites.js-old	2010-11-21 15:51:17.000000000 +0000
+++ sites.js	2010-11-21 16:07:11.000000000 +0000
@@ -377,6 +377,16 @@
 		charset: "utf-8",
 		extract: [['</a>?','<div']]
 	},
+  "letras.mus.br": {
+    charset: "iso-8859-1",
+    url: "http://letras.terra.com.br/winamp.php?musica={title}&artista={artist}",
+    urlFormat : [
+      {rep: "_", punct: "_@,;&\\/\"" },
+      {rep: "+", punct: " " },
+    ],
+    extract: [["</div>","</div>"]],
+    invalidIndicator: "Verifique se o nome do seu arquivo e sua",
+  },
 }
 
 //	--------------------------------------------------------------------------------------------------------------------------------
