Index: qxtglobalshortcut_x11.cpp
===================================================================
--- qxtglobalshortcut_x11.cpp	(revision 1208)
+++ qxtglobalshortcut_x11.cpp	(working copy)
@@ -74,8 +74,9 @@
         native |= ControlMask;
     if (modifiers & Qt::AltModifier)
         native |= Mod1Mask;
+    if (modifiers & Qt::MetaModifier)
+        native |= Mod4Mask;
     // TODO: resolve these?
-    //if (modifiers & Qt::MetaModifier)
     //if (modifiers & Qt::KeypadModifier)
     //if (modifiers & Qt::GroupSwitchModifier)
     return native;
