diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_core/com_trolltech_qt_core0.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_core/com_trolltech_qt_core0.cpp
index 23a7662..6ba7e74 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_core/com_trolltech_qt_core0.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_core/com_trolltech_qt_core0.cpp
@@ -4228,8 +4228,10 @@ QString  PythonQtWrapper_QDir::static_QDir_toNativeSeparators(const QString&  pa
 
 QString PythonQtWrapper_QDir::py_toString(QDir* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -4364,8 +4366,10 @@ qreal  PythonQtWrapper_QEasingCurve::valueForProgress(QEasingCurve* theWrappedOb
 
 QString PythonQtWrapper_QEasingCurve::py_toString(QEasingCurve* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -4412,8 +4416,10 @@ QEvent::Type  PythonQtWrapper_QEvent::type(QEvent* theWrappedObject) const
 
 QString PythonQtWrapper_QEvent::py_toString(QEvent* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << obj;
+#endif
   return result;
 }
 
@@ -7433,8 +7439,10 @@ QModelIndex  PythonQtWrapper_QModelIndex::sibling(QModelIndex* theWrappedObject,
 
 QString PythonQtWrapper_QModelIndex::py_toString(QModelIndex* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_core_builtin/com_trolltech_qt_core_builtin0.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_core_builtin/com_trolltech_qt_core_builtin0.cpp
index 07d8818..3ad258e 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_core_builtin/com_trolltech_qt_core_builtin0.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_core_builtin/com_trolltech_qt_core_builtin0.cpp
@@ -1250,8 +1250,10 @@ int  PythonQtWrapper_QLine::y2(QLine* theWrappedObject) const
 
 QString PythonQtWrapper_QLine::py_toString(QLine* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -1450,8 +1452,10 @@ qreal  PythonQtWrapper_QLineF::y2(QLineF* theWrappedObject) const
 
 QString PythonQtWrapper_QLineF::py_toString(QLineF* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -1870,8 +1874,10 @@ int  PythonQtWrapper_QPoint::y(QPoint* theWrappedObject) const
 
 QString PythonQtWrapper_QPoint::py_toString(QPoint* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -1996,8 +2002,10 @@ qreal  PythonQtWrapper_QPointF::y(QPointF* theWrappedObject) const
 
 QString PythonQtWrapper_QPointF::py_toString(QPointF* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -2321,8 +2329,10 @@ int  PythonQtWrapper_QRect::y(QRect* theWrappedObject) const
 
 QString PythonQtWrapper_QRect::py_toString(QRect* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -2665,8 +2675,10 @@ qreal  PythonQtWrapper_QRectF::y(QRectF* theWrappedObject) const
 
 QString PythonQtWrapper_QRectF::py_toString(QRectF* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -2936,8 +2948,10 @@ int  PythonQtWrapper_QSize::width(QSize* theWrappedObject) const
 
 QString PythonQtWrapper_QSize::py_toString(QSize* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -3077,8 +3091,10 @@ qreal  PythonQtWrapper_QSizeF::width(QSizeF* theWrappedObject) const
 
 QString PythonQtWrapper_QSizeF::py_toString(QSizeF* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui10.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui10.cpp
index 391341f..2b05be0 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui10.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui10.cpp
@@ -692,8 +692,10 @@ QTransform::TransformationType  PythonQtWrapper_QTransform::type(QTransform* the
 
 QString PythonQtWrapper_QTransform::py_toString(QTransform* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -7875,8 +7877,10 @@ qreal  PythonQtWrapper_QVector2D::y(QVector2D* theWrappedObject) const
 
 QString PythonQtWrapper_QVector2D::py_toString(QVector2D* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -8092,8 +8096,10 @@ qreal  PythonQtWrapper_QVector3D::z(QVector3D* theWrappedObject) const
 
 QString PythonQtWrapper_QVector3D::py_toString(QVector3D* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -8303,8 +8309,10 @@ qreal  PythonQtWrapper_QVector4D::z(QVector4D* theWrappedObject) const
 
 QString PythonQtWrapper_QVector4D::py_toString(QVector4D* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui2.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui2.cpp
index 796919d..63e376f 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui2.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui2.cpp
@@ -9427,8 +9427,10 @@ qreal  PythonQtWrapper_QGraphicsItem::zValue(QGraphicsItem* theWrappedObject) co
 
 QString PythonQtWrapper_QGraphicsItem::py_toString(QGraphicsItem* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << obj;
+#endif
   return result;
 }
 
@@ -12894,8 +12896,10 @@ return new PythonQtShell_QGraphicsObject(parent); }
 
 QString PythonQtWrapper_QGraphicsObject::py_toString(QGraphicsObject* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << obj;
+#endif
   return result;
 }
 
diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui4.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui4.cpp
index 077e294..d0fcd2f 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui4.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui4.cpp
@@ -4354,8 +4354,10 @@ int  PythonQtWrapper_QItemSelectionRange::width(QItemSelectionRange* theWrappedO
 
 QString PythonQtWrapper_QItemSelectionRange::py_toString(QItemSelectionRange* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -13872,8 +13874,10 @@ int  PythonQtWrapper_QMargins::top(QMargins* theWrappedObject) const
 
 QString PythonQtWrapper_QMargins::py_toString(QMargins* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui5.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui5.cpp
index 8a22514..cb77279 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui5.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui5.cpp
@@ -415,8 +415,10 @@ QMatrix4x4  PythonQtWrapper_QMatrix4x4::transposed(QMatrix4x4* theWrappedObject)
 
 QString PythonQtWrapper_QMatrix4x4::py_toString(QMatrix4x4* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -9467,8 +9469,10 @@ QPainterPath  PythonQtWrapper_QPainterPath::united(QPainterPath* theWrappedObjec
 
 QString PythonQtWrapper_QPainterPath::py_toString(QPainterPath* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -12792,8 +12796,10 @@ QPointF  PythonQtWrapper_QPolygonF::value(QPolygonF* theWrappedObject, int  i, c
 
 QString PythonQtWrapper_QPolygonF::py_toString(QPolygonF* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui6.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui6.cpp
index 415722c..4cb93f4 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui6.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui6.cpp
@@ -6690,8 +6690,10 @@ qreal  PythonQtWrapper_QQuaternion::z(QQuaternion* theWrappedObject) const
 
 QString PythonQtWrapper_QQuaternion::py_toString(QQuaternion* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui7.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui7.cpp
index 4890c63..4200fd0 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui7.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui/com_trolltech_qt_gui7.cpp
@@ -5000,8 +5000,10 @@ void PythonQtWrapper_QStyleOption::initFrom(QStyleOption* theWrappedObject, cons
 
 QString PythonQtWrapper_QStyleOption::py_toString(QStyleOption* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui_builtin/com_trolltech_qt_gui_builtin0.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui_builtin/com_trolltech_qt_gui_builtin0.cpp
index a743a34..cb99423 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui_builtin/com_trolltech_qt_gui_builtin0.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_gui_builtin/com_trolltech_qt_gui_builtin0.cpp
@@ -302,8 +302,10 @@ QTransform  PythonQtWrapper_QBrush::transform(QBrush* theWrappedObject) const
 
 QString PythonQtWrapper_QBrush::py_toString(QBrush* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -750,8 +752,10 @@ qreal  PythonQtWrapper_QColor::yellowF(QColor* theWrappedObject) const
 
 QString PythonQtWrapper_QColor::py_toString(QColor* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -2082,8 +2086,10 @@ QMatrix*  PythonQtWrapper_QMatrix::translate(QMatrix* theWrappedObject, qreal  d
 
 QString PythonQtWrapper_QMatrix::py_toString(QMatrix* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -2477,8 +2483,10 @@ qreal  PythonQtWrapper_QPen::widthF(QPen* theWrappedObject) const
 
 QString PythonQtWrapper_QPen::py_toString(QPen* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -3096,8 +3104,10 @@ QPoint  PythonQtWrapper_QPolygon::value(QPolygon* theWrappedObject, int  i, cons
 
 QString PythonQtWrapper_QPolygon::py_toString(QPolygon* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -3279,8 +3289,10 @@ QRegion  PythonQtWrapper_QRegion::xored(QRegion* theWrappedObject, const QRegion
 
 QString PythonQtWrapper_QRegion::py_toString(QRegion* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_network/com_trolltech_qt_network0.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_network/com_trolltech_qt_network0.cpp
index 11f8087..65ca21d 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_network/com_trolltech_qt_network0.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_network/com_trolltech_qt_network0.cpp
@@ -3770,8 +3770,10 @@ QByteArray  PythonQtWrapper_QNetworkCookie::value(QNetworkCookie* theWrappedObje
 
 QString PythonQtWrapper_QNetworkCookie::py_toString(QNetworkCookie* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
@@ -4418,8 +4420,10 @@ QString  PythonQtWrapper_QNetworkInterface::name(QNetworkInterface* theWrappedOb
 
 QString PythonQtWrapper_QNetworkInterface::py_toString(QNetworkInterface* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
diff --git a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_network/com_trolltech_qt_network1.cpp b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_network/com_trolltech_qt_network1.cpp
index 06d867b..61e2158 100644
--- a/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_network/com_trolltech_qt_network1.cpp
+++ b/3rdparty/pythonqt/generated_cpp/com_trolltech_qt_network/com_trolltech_qt_network1.cpp
@@ -88,8 +88,10 @@ bool  PythonQtWrapper_QSslKey::isNull(QSslKey* theWrappedObject) const
 
 QString PythonQtWrapper_QSslKey::py_toString(QSslKey* obj) {
   QString result;
+#ifdef QT_DEBUG
   QDebug d(&result);
   d << *obj;
+#endif
   return result;
 }
 
