Index: 3rdparty/pythonqt/src/PythonQtSignalReceiver.h
===================================================================
--- 3rdparty/pythonqt/src/PythonQtSignalReceiver.h	(revision 3364)
+++ 3rdparty/pythonqt/src/PythonQtSignalReceiver.h	(working copy)
@@ -99,7 +99,7 @@
 //! base class for signal receivers
 /*!
 */
-class PythonQtSignalReceiverBase : public QObject {
+class PYTHONQT_EXPORT PythonQtSignalReceiverBase : public QObject {
   Q_OBJECT
 public:
   PythonQtSignalReceiverBase(QObject* obj):QObject(obj) {};
@@ -108,7 +108,7 @@
 //! receives all signals for one QObject
 /*! we derive from our base but do not declare the QObject macro because we want to reimplement qt_metacall only.
 */
-class PythonQtSignalReceiver : public PythonQtSignalReceiverBase {
+class PYTHONQT_EXPORT PythonQtSignalReceiver : public PythonQtSignalReceiverBase {
 
 public:
   PythonQtSignalReceiver(QObject* obj);
