diff --git a/3rdparty/pythonqt/src/PythonQtConversion.cpp b/3rdparty/pythonqt/src/PythonQtConversion.cpp
index 1e3523f..0fbe291 100644
--- a/3rdparty/pythonqt/src/PythonQtConversion.cpp
+++ b/3rdparty/pythonqt/src/PythonQtConversion.cpp
@@ -674,7 +674,8 @@ QString PythonQtConv::PyObjGetString(PyObject* val, bool strict, bool& ok) {
     r = QString(PyString_AS_STRING(val));
   } else if (PyUnicode_Check(val)) {
 #ifdef WIN32
-    r = QString::fromUtf16(PyUnicode_AS_UNICODE(val));
+    r = QString::fromUtf16(
+        reinterpret_cast<const ushort*>(PyUnicode_AS_UNICODE(val)));
 #else
     PyObject *ptmp = PyUnicode_AsUTF8String(val);
     if(ptmp) {
@@ -1040,7 +1041,8 @@ PyObject* PythonQtConv::QStringToPyObject(const QString& str)
   } else {
 #ifdef WIN32
     //    return PyString_FromString(str.toLatin1().data());
-    return PyUnicode_FromUnicode(str.utf16(), str.length());
+    return PyUnicode_FromUnicode(
+        reinterpret_cast<const Py_UNICODE*>(str.utf16()), str.length());
 #else
     return PyUnicode_DecodeUTF16((const char*)str.utf16(), str.length()*2, NULL, NULL);
 #endif
