diff --git a/3rdparty/google-breakpad/client/linux/minidump_writer/linux_dumper.h b/3rdparty/google-breakpad/client/linux/minidump_writer/linux_dumper.h
index a4a3ab5..a46cbfc 100644
--- a/3rdparty/google-breakpad/client/linux/minidump_writer/linux_dumper.h
+++ b/3rdparty/google-breakpad/client/linux/minidump_writer/linux_dumper.h
@@ -50,7 +50,7 @@
 namespace google_breakpad {
 
 #if defined(__i386) || defined(__x86_64)
-typedef typeof(((struct user*) 0)->u_debugreg[0]) debugreg_t;
+typedef __typeof__(((struct user*) 0)->u_debugreg[0]) debugreg_t;
 #endif
 
 // Typedef for our parsing of the auxv variables in /proc/pid/auxv.
@@ -60,7 +60,7 @@ typedef Elf32_auxv_t elf_aux_entry;
 typedef Elf64_auxv_t elf_aux_entry;
 #endif
 
-typedef typeof(((elf_aux_entry*) 0)->a_un.a_val) elf_aux_val_t;
+typedef __typeof__(((elf_aux_entry*) 0)->a_un.a_val) elf_aux_val_t;
 
 // When we find the VDSO mapping in the process's address space, this
 // is the name we use for it when writing it to the minidump.
